text/0000-default-field-values.md

```diff
@@ -0,0 +1,1658 @@
+   The ability to specify default values would mean that `derivative` can
+   piggyback on the default value syntax due to this RFC. The mechanism for
+   removing or adding bounds can remain the same. Similar mechanisms could
+   also be added to the language itself.
+
+### `#[derive(SmartDefault)]`
+
+[`smart-default`]: https://crates.io/crates/smart-default
+
+The [`smart-default`] provides `#[derive(SmartDefault)]` custom derive macro.
+It functions similarly to `derivative` but is specialized for the `Default` trait.
+With it, you can write:
+
+
+```rust
+#[derive(SmartDefault)]
+struct RegexOptions {
+    #[default = "10 * (1 << 20)"]
+    size_limit: usize,
+    #[default = "2 * (1 << 20)"]
+    dfa_size_limit: usize,
+    #[default = true]
+    unicode: bool,
+}
+
+#[derive(SmartDefault)]
+enum Foo {
+    #[default]
+    Bar,
+    Baz,
+}
+```
+
++ The signal to noise ratio is still higher as compared to the notation in due
+  to this RFC. The problems aforementioned from the `derivative` crate with
+  respect to embedding Rust code in strings also persists.
+
++ Points 2-4 regarding `derivative` apply to `smart-default` as well.
+
++ The same syntax `#[default]` is used both by `smart-default` and by this RFC.
+  While it may seem that this RFC was inspired by `smart-default`, this is not
+  the case. Rather, this RFC's author came up with the notation independently.
+  That suggests that the notation is intuitive since and a solid design choice.
+
++ There is no trait `SmartDefault` even though it is being derived.
+  This works because `#[proc_macro_derive(SmartDefault)]` is in fact
+  not tied to any trait. That `#[derive(Serialize)]` refers to the same
+  trait as the name of the macro is from the perspective of the language's
+  static semantics entirely coincidental.
+
+  However, for users who aren't aware of this, it may seem strange that
+  `SmartDefault` should derive for the `Default` trait.
+
+### `#[derive(new)]`
+
+[`derive-new`]: https://crates.io/crates/derive-new
+
+The [`derive-new`] crate provides the `#[derive(new)]` custom derive macro.
+Unlike the two previous procedural macro crates, `derive-new` does not
+provide implementations of `Default`. Rather, the macro facilitates the
+generation of `MyType::new` constructors.
+
+For example, you may write:
+
+```rust
+#[derive(new)]
+struct Foo {
+    x: bool,
+    #[new(value = "42")]
+    y: i32,
+    #[new(default)]
+    z: Vec<String>,
+}
+
+Foo::new(true);
+
+#[derive(new)]
+enum Enum {
+    FirstVariant,
+    SecondVariant(bool, #[new(default)] u8),
+    ThirdVariant { x: i32, #[new(value = "vec![1]")] y: Vec<u8> }
+}
+
+Enum::new_first_variant();
+Enum::new_second_variant(true);
+Enum::new_third_variant(42);
+```
+
+Notice how `#[new(value = "vec![1]")`, `#[new(value = "42")]`,
+and `#[new(default)]` are used to provide values that are then omitted
+from the respective constructor functions that are generated.
```

Comment:
Stringly-typed again... ewwwww!

Reply:
Hopefully the proc macro system will be expanded to allow `attr = const-value` though. Is that the plan?